generated/nidaqmx/nidaqmx.proto

```diff
@@ -42,99 +88,1185 @@ enum NiDAQmxAttributes {
+  COUNTER_FREQUENCY_METHOD_DYN_AVG = 16065;
+}
+
+enum CurrentShuntResistorLocationWithDefault {
+  CURRENT_SHUNT_RESISTOR_LOCATION_WITH_DEFAULT_UNSPECIFIED = 0;
+  CURRENT_SHUNT_RESISTOR_LOCATION_WITH_DEFAULT_DEFAULT = -1;
+  CURRENT_SHUNT_RESISTOR_LOCATION_WITH_DEFAULT_INTERNAL = 10200;
+  CURRENT_SHUNT_RESISTOR_LOCATION_WITH_DEFAULT_EXTERNAL = 10167;
+}
+
+enum CurrentUnits2 {
+  CURRENT_UNITS2_UNSPECIFIED = 0;
+  CURRENT_UNITS2_AMPS = 10342;
+  CURRENT_UNITS2_FROM_CUSTOM_SCALE = 10065;
+}
+
+enum DigitalWidthUnits3 {
+  DIGITAL_WIDTH_UNITS3_UNSPECIFIED = 0;
+  DIGITAL_WIDTH_UNITS3_SECONDS = 10364;
+}
+
+enum Edge1 {
+  EDGE1_UNSPECIFIED = 0;
+  EDGE1_RISING = 10280;
+  EDGE1_FALLING = 10171;
+}
+
+enum EncoderType2 {
+  ENCODER_TYPE2_UNSPECIFIED = 0;
+  ENCODER_TYPE2_X_1 = 10090;
+  ENCODER_TYPE2_X_2 = 10091;
+  ENCODER_TYPE2_X_4 = 10092;
+  ENCODER_TYPE2_TWO_PULSE_COUNTING = 10313;
+}
+
+enum EncoderZIndexPhase1 {
+  ENCODER_Z_INDEX_PHASE1_UNSPECIFIED = 0;
+  ENCODER_Z_INDEX_PHASE1_A_HIGH_B_HIGH = 10040;
+  ENCODER_Z_INDEX_PHASE1_A_HIGH_B_LOW = 10041;
+  ENCODER_Z_INDEX_PHASE1_A_LOW_B_HIGH = 10042;
+  ENCODER_Z_INDEX_PHASE1_A_LOW_B_LOW = 10043;
+}
+
+enum ExcitationSource {
+  EXCITATION_SOURCE_UNSPECIFIED = 0;
+  EXCITATION_SOURCE_INTERNAL = 10200;
+  EXCITATION_SOURCE_EXTERNAL = 10167;
+  EXCITATION_SOURCE_NONE = 10230;
+}
+
+enum FrequencyUnits {
+  FREQUENCY_UNITS_UNSPECIFIED = 0;
+  FREQUENCY_UNITS_HZ = 10373;
+  FREQUENCY_UNITS_FROM_CUSTOM_SCALE = 10065;
+}
+
+enum FrequencyUnits2 {
+  FREQUENCY_UNITS2_UNSPECIFIED = 0;
+  FREQUENCY_UNITS2_HZ = 10373;
+}
+
+enum FrequencyUnits3 {
+  FREQUENCY_UNITS3_UNSPECIFIED = 0;
+  FREQUENCY_UNITS3_HZ = 10373;
+  FREQUENCY_UNITS3_TICKS = 10304;
+  FREQUENCY_UNITS3_FROM_CUSTOM_SCALE = 10065;
+}
+
+enum FuncGenType {
+  FUNC_GEN_TYPE_UNSPECIFIED = 0;
+  FUNC_GEN_TYPE_SINE = 14751;
+  FUNC_GEN_TYPE_TRIANGLE = 14752;
+  FUNC_GEN_TYPE_SQUARE = 14753;
+  FUNC_GEN_TYPE_SAWTOOTH = 14754;
+}
+
+enum GpsSignalType1 {
+  GPS_SIGNAL_TYPE1_UNSPECIFIED = 0;
+  GPS_SIGNAL_TYPE1_IRIGB = 10070;
+  GPS_SIGNAL_TYPE1_PPS = 10080;
+  GPS_SIGNAL_TYPE1_NONE = 10230;
+}
+
 enum GroupBy {
   option allow_alias = true;
   GROUP_BY_UNSPECIFIED = 0;
   GROUP_BY_GROUP_BY_CHANNEL = 0;
   GROUP_BY_GROUP_BY_SCAN_NUMBER = 1;
 }
 
-enum InputTermCfgWithDefault {
-  INPUT_TERM_CFG_WITH_DEFAULT_UNSPECIFIED = 0;
-  INPUT_TERM_CFG_WITH_DEFAULT_CFG_DEFAULT = -1;
-  INPUT_TERM_CFG_WITH_DEFAULT_RSE = 10083;
-  INPUT_TERM_CFG_WITH_DEFAULT_NRSE = 10078;
-  INPUT_TERM_CFG_WITH_DEFAULT_DIFF = 10106;
-  INPUT_TERM_CFG_WITH_DEFAULT_PSEUDO_DIFF = 12529;
+enum InputTermCfgWithDefault {
+  INPUT_TERM_CFG_WITH_DEFAULT_UNSPECIFIED = 0;
+  INPUT_TERM_CFG_WITH_DEFAULT_CFG_DEFAULT = -1;
+  INPUT_TERM_CFG_WITH_DEFAULT_RSE = 10083;
+  INPUT_TERM_CFG_WITH_DEFAULT_NRSE = 10078;
+  INPUT_TERM_CFG_WITH_DEFAULT_DIFF = 10106;
+  INPUT_TERM_CFG_WITH_DEFAULT_PSEUDO_DIFF = 12529;
+}
+
+enum LVDTSensitivityUnits1 {
+  L_V_D_T_SENSITIVITY_UNITS1_UNSPECIFIED = 0;
```

Comment:
Note: The proto generation mishandles ALLCAPS acronyms when converting to upper snake case. There are several shipping enums that are capitalized T_H_I_S_WAY. I'm working with @harsha-bhushan to figure out how to address this. I've created a task to track it and will include it in a separate PR.